### PR TITLE
Expose __type__ via SQL adapter

### LINKED ISSUE #6208: Expose `__type__` column in the SQL interface
When we are querying the inhviews, we can just resolve it to `__type__`. When not accessing subtypes, we'll need to inject the uuid directly.

### PR DESCRIPTION
**Pull request by aljazerzen**
Closes #6208

Exposes __type__ as an UUID columns of every table, including when using `FROM ONLY` (and we are compiling SQL to load data from tables that actually *do not* have the `__type__` column).

If using SQL adapter only, values of `__type__` are useless, as there is not way to query which type is associated with this id, as we don't expose `schema::ObjectType` (or anything from `schema` module) over SQL.

TODO:
- [x] add introspection

### CODE REVIEW

**Comment by aljazerzen** [comment]
@msullivan this is now ready

**Comment by msullivan** [comment]
> If using SQL adapter only, values of `__type__` are useless, as there is not way to query which type is associated with this id, as we don't expose `schema::ObjectType` (or anything from `schema` module) over SQL.
> 
Oh, lol. Maybe we should not have bothered with this.

File: edb/pgsql/resolver/relation.py [hunk 1]
```diff
@@ -385,16 +390,26 @@ def _construct_column(p: s_pointers.Pointer, ctx: Context) -> context.Column:
...
 
-    if isinstance(p, s_links.Link):
-        col.name = p.get_shortname(ctx.schema).name + '_id'
-
-        _, dbname = pgcommon.get_backend_name(ctx.schema, p, catenate=False)
-        col.reference_as = dbname
+    elif isinstance(p, s_links.Link):
+        if short_name.name == '__type__':
+            col.name = '__type__'
+            col.reference_as = '__type__'
```
Comment (msullivan) [inline edb/pgsql/resolver/relation.py:399]: Ugh, should we call it `__type__id`? That seems awful though

**Review by msullivan — APPROVED** [review]

File: edb/pgsql/resolver/relation.py (hunk 1 above)
Comment (aljazerzen) [reply edb/pgsql/resolver/relation.py:399]: It does look awful. It's a special-case column anyway, so we can change it into `the_id_of_the_type_of_this_object`, but I do think we can just leave it as `__type__`.

**Pull request merged**